03_letters/jonathanheaden/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func sortLetters(m map[rune]int) []string {
+	var rv = make([]string, 0)
+	var keys = make([]int, 0, len(m))
+	var sm = make(map[int]int)
+	for k := range m {
+		sm[int(k)] = m[k]
```

Comment:
This loop seems somewhat unnecessary, by my understanding it loops through the map `m` and then transfer the counts from `m` to int map `sm` along with int slice `keys`, after which point it sorts the `keys` slice. Then it fills out the string slice `rv` by looping through `keys` and appending to `rv` accordingly. 

While I'm not 100% certain I'm fairly sure that any non alphanumeric characters will break this, particularly characters from non-English alphabets and/or emojis.

Similarly though this could be simplified down to:
```
for k, count := range m {
  rs := string(k)
  rc := strconv.Itoa(count)
  rv = append(r, fmt.Sprintf(%s:%d", rs, rc))
}
sort.Strings(rv)
```